internal/pkg/controller/controller_test.go

```diff
@@ -0,0 +1,181 @@
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+// Creating a Controller to do a rolling upgrade upon updating the configmap or secret
+func TestControllerForUpdatingConfigmapShouldUpdateDeployment(t *testing.T) {
+	namespace := "test-reloader"
+	createNamespace(t, namespace)
+	defer deleteNamespace(t, namespace)
+
+	controller, err := NewController(client, "configMaps", namespace)
+	if err != nil {
+		logrus.Infof("Unable to create NewController error = %v", err)
+		return
+	}
+	stop := make(chan struct{})
+	defer close(stop)
+	go controller.Run(1, stop)
+	time.Sleep(10 * time.Second)
+
+	configmapName := configmapNamePrefix + "-update-" + randSeq(5)
+	configmapClient := client.CoreV1().ConfigMaps(namespace)
+	configmap := initConfigmap(namespace, configmapName)
+	configmap, err = configmapClient.Create(configmap)
+	if err != nil {
+		panic(err)
+	}
+	logrus.Infof("Created Configmap %q.\n", configmap.GetObjectMeta().GetName())
+	time.Sleep(10 * time.Second)
+
+	logrus.Infof("Updating Configmap %q.\n", configmap.GetObjectMeta().GetName())
+	configmap, err = configmapClient.Get(configmapName, metav1.GetOptions{})
+	if err != nil {
+
+	}
+	configmap = updateConfigmap(namespace, configmapName)
+	_, updateErr := configmapClient.Update(configmap)
+
+	// TODO: Add functionality to verify reloader functionality here
+
+	if updateErr != nil {
+		controller.client.CoreV1().ConfigMaps(namespace).Delete(configmapName, &metav1.DeleteOptions{})
+		panic(updateErr)
+	}
+	time.Sleep(10 * time.Second)
+	logrus.Infof("Deleting Configmap %q.\n", configmap.GetObjectMeta().GetName())
+	controller.client.CoreV1().ConfigMaps(namespace).Delete(configmapName, &metav1.DeleteOptions{})
+	time.Sleep(15 * time.Second)
+}
+
+func TestControllerForUpdatingSecretShouldUpdateDeployment(t *testing.T) {
+	namespace := "test-reloader-secrets"
+	createNamespace(t, namespace)
+	defer deleteNamespace(t, namespace)
+
+	controller, err := NewController(client, "secrets", namespace)
+	if err != nil {
+		logrus.Infof("Unable to create NewController error = %v", err)
+		return
+	}
+	stop := make(chan struct{})
+	defer close(stop)
+	go controller.Run(1, stop)
+	time.Sleep(10 * time.Second)
+
+	secretName := secretNamePrefix + "-update-" + randSeq(5)
+	secretClient := client.CoreV1().Secrets(namespace)
+	secret := initSecret(namespace, secretName)
+	secret, err = secretClient.Create(secret)
+	if err != nil {
+		panic(err)
+	}
+	logrus.Infof("Created Secret %q.\n", secret.GetObjectMeta().GetName())
+	time.Sleep(10 * time.Second)
+
+	logrus.Infof("Updating Secret %q.\n", secret.GetObjectMeta().GetName())
+	secret, err = secretClient.Get(secretName, metav1.GetOptions{})
+	if err != nil {
+
+	}
+	secret = updateSecret(namespace, secretName)
+	_, updateErr := secretClient.Update(secret)
+
+	// TODO: Add functionality to verify reloader functionality here
+
+	if updateErr != nil {
+		controller.client.CoreV1().Secrets(namespace).Delete(secretName, &metav1.DeleteOptions{})
```

Comment:
Error return value of `(github.com/stakater/Reloader/vendor/k8s.io/client-go/kubernetes/typed/core/v1.SecretInterface).Delete` is not checked